install-mac-dev.sh

```diff
@@ -9,25 +9,37 @@ if [[ $# -gt 0 ]]; then
     exit 1
 fi
 
-brew install leveldb || brew upgrade leveldb
+# Create and activate a local Python 3.6 venv
+python3.6 -m venv venv
+source venv/bin/activate
+pip install --upgrade pip
+
+# Install all requirements except plyvel
+cat requirements.txt | grep -v plyvel > requirements.mac.txt
+CFLAGS='-mmacosx-version-min=10.7 -stdlib=libc++' pip install -r requirements.mac.txt
+rm requirements.mac.txt
 
-# Make sure we build plyvel properly to work with the installed leveldb
-CFLAGS='-std=c++11' pip install --user --force-reinstall --ignore-installed --no-binary :all: plyvel
+# A recent version of leveldb is required
+brew install leveldb || brew upgrade leveldb
 
-pip install -U -r requirements.txt
+# Make sure we build plyvel properly to work with the installed leveldb on recent OSX versions
+CFLAGS='-mmacosx-version-min=10.7 -stdlib=libc++ -std=c++11' pip install plyvel
+# --force-reinstall --ignore-installed --no-binary :all:
```

Comment:
I am pretty sure we need the --no-binary flag here, otherwise a cached binary might be used, which renders the CFLAGS moot as well. I'll try this locally.